paper/paper.tex

```diff
@@ -0,0 +1,268 @@
+\\
+
+Many software packages exist for survey analysis\footnote{A comprehensive list is provided by \cite{SummarySurveyAnalysis}}. Notable examples include the R survey package, SAS/STAT, SPSS Complex Samples, Stata, and SUDAAN. The R survey package by Thomas Lumley\cite{lumley2004analysis} is widely recognized for its comprehensive capabilities and open-source availability. However, it is limited by R's computational efficiency, especially for large-scale data. Survey.jl leverages Julia to offer a faster resampling framework for variance estimation and survey data analysis.
+
+%% Short summary of the paper
+
+% \section{Related work}
+
+% %% Check links. It's from here: https://www.hcp.med.harvard.edu/statistics/survey-soft/#Packages
+
+% There are many packages for survey analysis. A list and summary of the packages is provided by Section on Survey Research Methods, American Statistical Association \cite{SummarySurveyAnalysis}. 
+
+% \href{https://www.hcp.med.harvard.edu/statistics/survey-soft/am.html}{AM Software}, 
+% \href{https://www.hcp.med.harvard.edu/statistics/survey-soft/bascula.html}{Bascula}, 
+% \href{https://www.hcp.med.harvard.edu/statistics/survey-soft/cenvar.html}{CENVAR}, 
+% \href{https://www.hcp.med.harvard.edu/statistics/survey-soft/clusters.html}{CLUSTERS},  
+% \href{https://www.cdc.gov/epiinfo/index.html}{Epi Info},  
+% \href{https://www.statcan.gc.ca/eng/survey/methodology/Generalized_Estimation_System-eng.htm}{Generalized Estimation System (GES)},  
+% \href{https://isr.umich.edu/}{IVEware},  
+% \href{https://catalog.iastate.edu/azcourses/stat/}{PCCARP},  
+% \href{https://cran.r-project.org/package=survey}{R survey package}, 
+% \href{https://www.sas.com/en_us/home.html}{SAS/STAT},  
+% \href{https://www.ibm.com/products/spss-statistics}{SPSS Complex Samples},  
+% \href{https://www.stata.com/}{Stata},  
+% \href{https://sudaanorder.rti.org/}{SUDAAN},  
+% \href{https://www.census.gov/data/software/vplx.html}{VPLX},  
+% \href{https://www.westat.com/wesvar/}{WesVar}
+
+% The survey package in R by Thomas Lumely \cite{lumley2004analysis} is the widely used open-source package. 
+
+\section{Survey design}
+
+A \verb|SurveyDesign| object can be created to incorporate the sampling design. This object requires the following parameters: \verb|data::DataFrame|, which is the survey data in the form of a \verb|DataFrame|; \verb|clusters::Symbol|, specifying the column name containing the clusters; \verb|strata::Symbol|, specifying the column name containing the strata; \verb|weights::Symbol|, indicating the column name containing the sampling weights; and \verb|popsize::Symbol|, indicating the column name containing the population size.\footnote{Internaly, there is a single constructor for all types of surveys. Every survey is assumed to be a complex survey. If there is no stratification, we assume that everything is part of one stratum. If there is no clustering, we assume each member is a cluster.}  
```

Comment:
`This object requires the following parameters`: instead of `parameters`, should we say `properties` instead? Usually parameters are constants/values that tag some method/type, or I like to think of them as tunable values associated to some model.

Reply:
How about `argument`?

Reply:
`argument` definitely sounds better.

Reply:
> How about `argument`?

```suggestion
A \verb|SurveyDesign| object can be created to incorporate the sampling design. This object requires the following arguments: \verb|data::DataFrame|, which is the survey data in the form of a \verb|DataFrame|; \verb|clusters::Symbol|, specifying the column name containing the clusters; \verb|strata::Symbol|, specifying the column name containing the strata; \verb|weights::Symbol|, indicating the column name containing the sampling weights; and \verb|popsize::Symbol|, indicating the column name containing the population size.\footnote{Internaly, there is a single constructor for all types of surveys. Every survey is assumed to be a complex survey. If there is no stratification, we assume that everything is part of one stratum. If there is no clustering, we assume each member is a cluster.}  
```